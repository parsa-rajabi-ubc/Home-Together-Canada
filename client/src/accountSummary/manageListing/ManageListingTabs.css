.manage-listing-tabs {
    @apply mx-4 rounded-full shadow-md border-2 w-1/3 ;
}

.manage-listing-tabs:focus {
    @apply outline-none;
}

.unselected-tab {
    @apply border-green-300;
}