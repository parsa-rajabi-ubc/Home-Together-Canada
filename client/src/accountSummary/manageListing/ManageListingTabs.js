/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.6.10
 *
 * @Description: Manage Listing Tabs
 *
 */

import React, {useState, useEffect} from "react";
import PropTypes from "prop-types";

export const MANAGE_LISTING_TAB_TEXT = {
    TITLE: "Manage Listings",
    DESCRIPTION: "Use the tabs below to view your listings based on their status.",
    LIVE: "Live Listings",
    INACTIVE: "Inactive Listings",
    NO_LISTING: "No Listings"
}

export const TAB_LABELS = [
    "Live",
    "Inactive",
];

function ManageListingTabs(props) {
    const {
        setActiveTab,
    } = props;
    // initialize current tab to 0 ie. live listings
    const [currentTab, setCurrentTab] = useState();

    // useEffects
    useEffect(() => {
        setActiveTab(currentTab);
    }, [currentTab])



    const tabs = TAB_LABELS.map(
        (tab) =>
            <button key={tab} type={"button"}
                    className={tab === currentTab ? "sub-page-items selected w-1/3 rounded-full shadow-md border-2 focus:outline-none" : "sub-page-items w-1/3 rounded-full shadow-md border-2 border-green-300 focus:outline-none"}
                    onClick={() => setCurrentTab(tab)}>
                {tab + " Listings"}
            </button>
    );

    return (
        <div>
            <h3 className={"account-summary-info-header"}> {MANAGE_LISTING_TAB_TEXT.TITLE} </h3>
            <p className="account-summary-info-text"> {MANAGE_LISTING_TAB_TEXT.DESCRIPTION} </p>
            <div className={"text-center mx-auto my-10"}>
                {tabs}
            </div>
        </div>
    );
}

ManageListingTabs.propTypes = {
    setActiveTab: PropTypes.func,
};

export default ManageListingTabs;