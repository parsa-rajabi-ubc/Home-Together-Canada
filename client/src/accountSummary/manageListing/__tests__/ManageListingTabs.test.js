/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.06.13
 *
 * @Description: Manage Listing Test
 *
 */
import React from 'react';
import renderer from 'react-test-renderer'
import ManageListingTabs from "../ManageListingTabs";
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));


describe('ManageListingTabs', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties (Member)", () => {
            // given
            const props = {
                setActiveTab: jest.fn(),
                accountType: 'member'
            }

            // when
            const component = renderer.create(<Router><ManageListingTabs {...props} /></Router>).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
        it("should render correctly regardless of properties (Business)", () => {
            // given
            const props = {
                setActiveTab: jest.fn(),
                accountType: 'business'
            }

            // when
            const component = renderer.create(<Router><ManageListingTabs {...props} /></Router>).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    })
})