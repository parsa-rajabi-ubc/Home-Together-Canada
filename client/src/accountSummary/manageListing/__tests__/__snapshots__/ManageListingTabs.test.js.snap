// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageListingTabs Snapshot test should render correctly regardless of properties (Business) 1`] = `
<div>
  <h3
    className="account-summary-info-header"
  >
     
    Manage Listings
     
  </h3>
  <p
    className="account-summary-info-text"
  >
     
    Use the tabs below to view your listings based on their status.
     
  </p>
  <div
    className="text-center mx-auto my-10"
  >
    <button
      className="w-1/5 sub-page-items selected manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Live Listings
    </button>
    <button
      className="w-1/5 sub-page-items unselected-tab manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Inactive Listings
    </button>
    <button
      className="w-1/5 sub-page-items unselected-tab manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Rejected Listings
    </button>
    <button
      className="w-1/5 sub-page-items unselected-tab manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Pending Listings
    </button>
  </div>
</div>
`;

exports[`ManageListingTabs Snapshot test should render correctly regardless of properties (Member) 1`] = `
<div>
  <h3
    className="account-summary-info-header"
  >
     
    Manage Listings
     
  </h3>
  <p
    className="account-summary-info-text"
  >
     
    Use the tabs below to view your listings based on their status.
     
  </p>
  <div
    className="text-center mx-auto my-10"
  >
    <button
      className="false sub-page-items selected manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Live Listings
    </button>
    <button
      className="false sub-page-items unselected-tab manage-listing-tabs"
      onClick={[Function]}
      type="button"
    >
      Inactive Listings
    </button>
  </div>
</div>
`;
